pkg/tbtc/chain.go

```diff
@@ -26,6 +29,28 @@ type DistributedKeyGenerationChain interface {
 	OnDKGStarted(
 		func(event *DKGStartedEvent),
 	) subscription.EventSubscription
+
+	// OnDKGResultSubmitted registers a callback that is invoked when an on-chain
+	// notification of the DKG result submission is seen.
+	OnDKGResultSubmitted(
+		func(event *DKGResultSubmittedEvent),
+	) subscription.EventSubscription
+
+	// SubmitDKGResult submits the DKG result to the chain, along with signatures
+	// over result hash from group participants supporting the result.
+	SubmitDKGResult(
+		memberIndex group.MemberIndex,
+		result *dkg.Result,
+		signatures map[group.MemberIndex][]byte,
+	) error
+
+	// TODO: Check what should be the argument for this function.
+	// IsDKGResultSubmitted checks whether the DKG result was already submitted.
+	IsDKGResultSubmitted(groupPublicKeyBytes []byte) (bool, error)
```

Comment:
I'm wondering whether we can replace that function with a function that will return the current DKG state. The `WalletRegistry` contract exposes `getWalletCreationState` function. I think we can introduce a `GetDKGState` function to the `DistributedKeyGenerationChain` that will return an enum value corresponding to DKG states and will call contract's `getWalletCreationState` underneath in the future. Then, we can use this function to check whether the result was already submitted in the `tbtc.dkgResultSubmitter`. WDYT?

Reply:
Done in 59752ac2bc14d0b6dc8074c2d3a510647652188f.

However, adding `GetDKGState() (DKGState, error)` to the `DistributedKeyGenerationChain` interface forces all the 
implementations of that interface to use the same enums.
If we wanted to use a different chain than ethereum-based one, we are forced to use the same DKG states, which could be a problem.

Returning a bool would be probably more generic, because we don't care about DKG states.

Reply:
> However, adding GetDKGState() (DKGState, error) to the DistributedKeyGenerationChain interface forces all the implementations of that interface to use the same enums.
If we wanted to use a different chain than ethereum-based one, we are forced to use the same DKG states, which could be a problem.

This is good. The underlying chain has nothing to do with DKG states. This is something specific for tBTC contracts and should be followed by all chains. Also, the client will control DKG challenges and approvals so it must be aware of DKG states.